rdm/audit_for_gaps.py

```diff
@@ -175,7 +178,69 @@ def _sorted_checklist_items(unsorted_checklist):
         key = checklist_item.get('reference', '')
         description = checklist_item.get('description', '')
         unsorted_items.append(key + ' ' + description)
-    return sorted(unsorted_items)
+    # The use of the SectionalAnalysis tool ensures '62304:5.1.2' < '62304:5.1.11' < '62304:5.2.1'
+    return sorted(unsorted_items, key=SectionalAnalysis)
+
+
+class SectionalAnalysis:
```

Comment:
@orwonthe I haven't read through the sorting implementation, so this may be irrelevant, but python's tuples have a nice property that they can be sorted:

```
>>> tuple() < (1,)
True
>>> (1,) < (2,)
True
>>> (2,) < (2,1)
True
>>> (2,1) < (2,2)
True
>>> (2,2) < (2,2,1)
True
>>> (2,2,1) < (3,)
```

Perhaps this could be used to simplify the sorting implementation somehow?